### DISCUSSION TITLE
Add a function to import GPX file directly

### DISCUSSION THREAD

**Discussion by marunguy** [organicmaps/organicmaps#372] [Ideas]
Currently, omaps can only import kml file.
It is a little inconvenient to convert a gpx file to a kml file and to import in somecase(without internet).
Is it possible to add a function to import GPX file directly?

**Comment by biodranik** [comment]
That is a great idea, of course, we can add some GPX parser.

**Comment by rtsisyk** [comment]
You can try https://maps.omaps.app/ to convert GPX/GeoJSON into KML.

**Comment by ooprizrakoo** [comment]
https://github.com/organicmaps/organicmaps/issues/624

**Comment by 0709wiwiwi** [comment]


**Reply by biodranik** [reply]
The same functionality is already implemented with KML and KMZ formats.

GPX support is planned.

**Reply by IzzySoft** [reply]
No pressing, but is there an approximate ETA yet? I'm about to switch over from OsmAnd, and one thing I often use is importing a GPX track to follow it (so ideally I can use that as "navigation route" – but even being able to see it on the map would already help).

**Reply by biodranik** [reply]
The task itself is about a week of work for a senior C++ developer. It is one of our top priority tasks. No promises yet, because we work in our free time, and focus first on the most critical bugs.

**Reply by biodranik** [reply]
Please follow #624 for updates.

**Reply by IzzySoft** [reply]
Thanks a lot, @biodranik! Of course I didn't demand (or even expect) any promises. I'm aware of the volunteer work (disclosure: I'm one of the F-Droid maintainers, so I'm well aware what that means).

And thanks for the pointer, I've just subscribed to that issue. Looking forward to it, hoping it will include track imports as well (the ones I mostly import are recordings from [Tower Collector](https://f-droid.org/en/packages/info.zamojski.soft.towercollector/)). Until then I'll enjoy the clean interface of a well working OSM app; thanks for that! :heart_eyes:

**Reply by biodranik** [reply]
You may help by providing examples of some rich-featured GPX tracks you want to support, and by mapping their corresponding features to the KML format. I assume that GPX should perfectly work as a subset of KML.

**Reply by biodranik** [reply]
And btw thanks for the F-Droid! You definitely need a lot of support, especially related to the slow build process :) But that's another issue.

**Reply by IzzySoft** [reply]
> especially related to the slow build process

Guess your last look was more than a week ago. We got new build servers :smiley: :partying_face: Related issues will be addressed as well, but things take time (we all hope: not too much of that :see_no_evil:)

As for the GPX files: Shall I attach an example here, or at the other issue? I'll unfortunately be unable to perform the mapping as I'm not deep enough into that topic…

**Reply by biodranik** [reply]
It's better to attach them to that another issue, to have everything in one place.

**Reply by IzzySoft** [reply]
[Done](https://github.com/organicmaps/organicmaps/issues/624#issuecomment-1289764862), thanks!
